lib/rubocop/cop/rspec/subject_stub.rb

```diff
@@ -56,25 +58,19 @@ class SubjectStub < Cop
         #     expect(foo).to receive(:bar).with(1)
         #     expect(foo).to receive(:bar).with(1).and_return(2)
         #
-        #   @example source that not matches
-        #     expect(foo).to all(receive(:bar))
-        #
         def_node_matcher :message_expectation?, <<-PATTERN
-          {
-            (send nil? :allow (send nil? %))
-            (send (send nil? :expect (send nil? %)) :to #expectation?)
-          }
+          (send
+            (send nil? { :expect :allow } (send nil? %))
+            #{Runners::ALL.node_pattern_union}
+            #message_expectation_matcher?
+          )
         PATTERN
 
-        def_node_matcher :all_matcher?, '(send nil? :all ...)'
-
-        def_node_search :receive_message?, '(send nil? :receive ...)'
-
-        def expectation?(node)
-          return if all_matcher?(node)
-
-          receive_message?(node)
-        end
+        def_node_search :message_expectation_matcher?, <<-PATTERN
+          (send nil? {
+            :receive :receive_messages :receive_message_chain :have_received
```

Comment:
this also might play nice in `Language`

Reply:
I was thinking about this, but couldn't find another usage that would benefit from that extraction.
Just noticed that `stub_chain` (an alias of `receive_message_chain`?) is supported in a number of cops, even though it's not in RSpec 3 anymore. Do you think it makes sense to add it to this cop?

Reply:
Since it requires monkey patching, and we defined as a no-goal to enforce expect over should, I think we should not take care of any of the old syntax methods and clean up them.